packages/git/src/browser/git-widget.tsx

```diff
@@ -391,118 +411,119 @@ export class GitWidget extends VirtualWidget implements StatefulWidget {
         return [username, email];
     }
 
-    protected renderGitItemButtons(repository: Repository, change: GitFileChange): h.Child {
-        const buttons: h.Child[] = [];
-        if (change.staged) {
-            buttons.push(h.a({
-                className: 'toolbar-button',
-                title: 'Unstage Changes',
-                onclick: async event => {
-                    try {
-                        await this.git.unstage(repository, change.uri, { treeish: 'HEAD', reset: 'index' });
-                    } catch (error) {
-                        this.logError(error);
-                    }
-                }
-            }, h.i({ className: 'fa fa-minus' })));
+    protected unstageHandler = async (repository: Repository, change: GitFileChange) => {
+        try {
+            await this.git.unstage(repository, change.uri);
+        } catch (error) {
+            this.logError(error);
+        }
+    }
+
+    protected discardHandler = async (repository: Repository, change: GitFileChange) => {
+        const options: Git.Options.Checkout.WorkingTreeFile = { paths: change.uri };
+        if (change.status === GitFileStatus.New) {
```

Comment:
Without trying it out, I can see, you have overlooked something during the rebase:
```
// Allow deletion, only iff the same file is not yet in the Git index.
if (await this.git.lsFiles(repository, change.uri, { errorUnmatch: true })) {
```

From: https://github.com/theia-ide/theia/pull/2129/files#diff-b87157e6670b5ef4be69ade938256e16L413

The whole change is here: https://github.com/theia-ide/theia/pull/2100/files#diff-392c37f87fa84f7102bfb5248b4837d8